### DISCUSSION TITLE
Decentralized technologies

### DISCUSSION THREAD

**Discussion by jefftangx** [athensresearch/athens#603] [General]
#9 is for backends. I'd like to separate decentralized and centralized technologies for now. Here's a big list of decentralized projects I've heard a little about. Some are storage related, some are not.

- [OrbitDB](https://github.com/orbitdb/orbit-db) - db
- [Gun](https://gun.eco/) - db
- [3box](https://3box.io/) - storage
- [fission.codes](https://fission.codes/) - storage
- [SourceCred](https://sourcecred.io/) - credit system
- [TheGraph](https://thegraph.com/) - data
- [Underlay](https://underlay.mit.edu/) - public knowledge graph
- [Solid Project](https://solidproject.org/)
- [Mirror.xyz](https://mirror.xyz/) - publishing
- [radicle.xyz](https://radicle.xyz/)
- [uprtcl.io](http://uprtcl.io/)
- [AnyType](https://anytype.io/) - notetaking app
- [Fairmint](https://fairmint.co/) - fundraising
- [Gitcoin](https://gitcoin.co/) - pay outs
- [Scuttlebutt](https://scuttlebutt.nz/) - social network
- [Spritely](https://spritelyproject.org/) - social web

**Comment by oogetyboogety** [comment]
TheGraph can host defi exchange of datasets using a standard called erasure protocol

**Comment by oogetyboogety** [comment]
The examples use IPFS but could just use IPLD on top of IPFS

**Comment by seltzered** [comment]
holochain (folks behind hylo.com are eventually migrating to it I think)

**Comment by travis** [comment]
we were just starting to think about implementing something very similar to Athens on top of Solid at @itme - would love to chat!

**Comment by gturkoglu** [comment]
- [arweave.org](https://www.arweave.org) - storage
You pay once to store data *permanently*.

**Comment by Tamriel** [comment]
I propose using a Conflict-free Replicated Data Type (CRDT). The library [Automerge](https://github.com/automerge/automerge) describes the concept quite well:

> You can have a copy of the application state locally on several devices (which may belong to the same user, or to different users). Each user can independently update the application state on their local device, even while offline, and save the state to local disk.
When a network connection is available, Automerge figures out which changes need to be synced from one device to another, and brings them into the same state.
If the state was changed concurrently on different devices, Automerge automatically merges the changes together cleanly, so that everybody ends up in the same state, and no changes are lost. There are no merge conflicts to resolve.

The network communication is handled by a separate layer, e.g. by a peer-to-peer log store like Scuttlebutt.

The libraries Automerge and Yjs already have some real world usages, listed at the bottom of this site about [CRDT implementations](https://crdt.tech/implementations).

A new interesting implementation is the [Fluid Framework](https://fluidframework.com/docs/).

**Comment by behaviary** [comment]
Hey, I was asked to contribute here by @tangjeff0. Wanted to highlight a project that I've been working on which is a new blockchain.

I'm sure you all can find the marketing website on your own, thus here are the [developer docs for NEAR protocol](https://docs.near.org/docs/develop/basics/getting-started). 

This is a layer 1, meaning the infrastructure layer, blockchain. It has an account system as well as a storage solution, but I would recommend something like [IPFS](https://ipfs.io/) or [Ceramic](https://ceramic.network/) for storage. 

For this use case, you could make the permissioned account system permissionless. Meaning that people can bring their accounts and associated data with them without needing to go through a server.
